### PR TITLE
test(headles-react): add test for ssr-commerce

### PR DESCRIPTION
**Pull request by y-lakhdar**
* Copied over and adjusted the tests from `ssr` to `ssr-commerce`
* Add missing export from headless

https://coveord.atlassian.net/browse/KIT-3696

### CODE REVIEW

File: packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    expect(typeof controllers.useProductList).toEqual('function');
+    expect(typeof controllers.useSearchBox).toEqual('function');
+  });
+
+  describe('context providers', () => {
+    const productItemTestId = 'product-item';
+    const numProducts = 10;
+    const engineDefinition = defineCommerceEngine({
+      configuration: sampleConfig,
+      // TODO: Generalize tests to test all defined controllers dynamically
```
Comment (fbeaudoincoveo) [inline packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx:110]: We should mention a Jira in this comment

File: packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        test('should return engine with hydrated state provider', async () => {
+          const {result} = renderHook(() => useEngine(), {
+            wrapper: hydratedStateProviderWrapper,
+          });
+          expect(result.current).toStrictEqual(hydratedState.engine);
+        });
+      });
+
+      describe('controller hooks', () => {
+        // TODO: Generalize to loop through all defined controllers dynamically
```
Comment (fbeaudoincoveo) [inline packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx:287]: Add Jira

File: packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        describe('with HydratedStateProvider', () => {
+          test('should define both state and controller', () => {
+            const {result} = renderHook(() => useSearchBox(), {
+              wrapper: hydratedStateProviderWrapper,
+            });
+            expect(result.current.state).toBeDefined();
+            expect(result.current?.controller).toBeDefined();
+          });
+
+          // TODO(DEBUG): hydratedState might need to be passed into the wrapper
```
Comment (fbeaudoincoveo) [inline packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx:308]: Add Jira

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by louis-bompart — APPROVED** [review]

File: packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+  const sampleConfig = getSampleCommerceEngineConfiguration();
+  beforeEach(() => {
+    errorSpy = vi.spyOn(console, 'error');
+  });
+
+  afterEach(() => {
+    errorSpy.mockClear();
+  });
+
+  test('defines react search engine', () => {
```
Comment (alexprudhomme) [inline packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx:45]:
```suggestion
  test('defines react commerce engine', () => {
```

File: packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx (hunk 1 above)
Comment (alexprudhomme) [reply packages/headless-react/src/ssr-commerce/commerce-engine.test.tsx:110]: This is from search-engine.test.tsx, we should remove the comment imo

**Review by alexprudhomme — COMMENTED** [review]
I think we should address those TODOs now and decide whether they are worth doing. If not, we should remove them from both /ssr and /ssr-commerce.

**Pull request merged**